### PR TITLE
Harmonize request titles in json schemas

### LINKED ISSUE #1880: Possible mismatch between schema entity property and API return type
When developing a client in Rust I noticed a mismatch in the generated model between expected an actual value retrieved from the API. Specifically the `updatedAt` property in the `Database` model is specified at ISO time string, but the return value is unix epoch / i64.

### PR DESCRIPTION
**Pull request by roeap**
### Describe your changes :

- Updated some titles in schema files to generate more consistent type names and classes.
- Run prettier on all json schema files

Some minor fixes to enable type generation for rust client.
- fix `required` names in `webhook.json`
- removed `filters` field from required in `rule.json`, since hat field has been removed. Should there be another required prop from the new `filterRegex` etc...
- removed all fields from `required` in `databaseConnectionConfig.json` since none of the fileds are defined - should there be something now?

closes #1880 - while this issue has since been addressed in other PRs (i.e. using EPOCH for timestamps), this PR addresses some comments in the thread.

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@sureshms @harshach

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json
Comment (sureshms) [inline catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json:105]: Do you intend to change this to dateTime type?

File: catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json
Comment (sureshms) [inline catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json:52]: Is this change from timestamp to dateTime intentional?

File: catalog-rest-service/src/main/resources/json/schema/entity/services/pipelineService.json
Comment (sureshms) [inline catalog-rest-service/src/main/resources/json/schema/entity/services/pipelineService.json:55]: Intentional change?

File: catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json
Comment (roeap) [reply catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json:52]: sorry, it seems i made some error merging this. The migration to timestamp was the exact reason i asked in the first place :). reverting ...

**Comment by sureshms** [comment]
@roeap, thank you for the patch. Posted some comments in the patch.

**Comment by pmbrull** [comment]
@harshach when merging this in we'll need to update the imports in the `ingestion` to match with the new `title`s

File: catalog-rest-service/src/main/resources/json/schema/entity/services/pipelineService.json
Comment (roeap) [reply catalog-rest-service/src/main/resources/json/schema/entity/services/pipelineService.json:55]: no - fixed it.

File: catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json
Comment (roeap) [reply catalog-rest-service/src/main/resources/json/schema/entity/tags/tagCategory.json:105]: see above

**Comment by roeap** [comment]
@sureshms - thanks for the very quick review, and sorry for the errors merging the most recent schema. Addressed all your comments.

**Comment by harshach** [comment]
Thanks @roeap for the PR. This requires changing the Python code quite a bit as @pmbrull pointed. We are doing release this week and will merge this post release.

**Comment by roeap** [comment]
@harshach - absolutely, also I'm very much looking forward to the next release :).

To support the effort, I ran a type checker through the code and replaces all erroneous imports. For some reason the local docker setup script failed, so I could not run integration tests.

**Comment by harshach** [comment]
> @harshach - absolutely, also I'm very much looking forward to the next release :).
> 
> To support the effort, I ran a type checker through the code and replaces all erroneous imports. For some reason the local docker setup script failed, so I could not run integration tests.

Thanks @roeap I think the integration tests setup is running into some other issue but yes will merge it this week

**Comment by roeap** [comment]
@harshach @pmbrull - I merged latest main to resolve conflicts with the latest changes in the json files. Looking at the failures in the python tests it seems that after switching to py3.9 in CI, the `make`generate` script no longer works due to an exception raised in the `datamodel-code-generator` package. specifically.

```sh
Traceback (most recent call last):
  File "/home/robstar/github/OpenMetadata/env39/bin/datamodel-codegen", line 8, in <module>
    sys.exit(main())
  File "/home/robstar/github/OpenMetadata/env39/lib/python3.9/site-packages/datamodel_code_generator/__main__.py", line 432, in main
    pyproject_toml_path = root / "pyproject.toml"
TypeError: unsupported operand type(s) for /: 'tuple' and 'str'
```

Locally I am able to reproduce the behaviour - i.e. running generate in 3.8 works, but fails in 3.9. HAve you observed this before as well and are maybe aware of a fix I need to apply?

Also it seems when installing dev and test dependencies there is a version conflict for the click package between black and flask - but I was able to resolve that by forcing the latest click version.

Any ideas?

**Comment by roeap** [comment]
It seems pinning the black version as in #2499 resolves the issue in the generate script. Will merge once that PR lands.

**Comment by pmbrull** [comment]
> It seems pinning the black version as in #2499 resolves the issue in the generate script. Will merge once that PR lands.

Exactly @roeap, we had this same issue in other open PRs as well. Really appreciate the help here 🙏  Many thanks!

**Comment by roeap** [comment]
I took the liberty and added prettier as a pre-commit hook as well - so far limited to the schema files. Personally I found this makes working with large amounts of json files much easier. Of course I'll remove this if you don't want to go this way in your repo.

**Comment by harshach** [comment]
Thanks @roeap .
@pmbrull are there anything else do we need from this PR or is it good to me be merged. I assume the py-tests are failing because of lib version?

**Comment by pmbrull** [comment]
> 

Hi @harshach, opened https://github.com/open-metadata/OpenMetadata/pull/2500 as well to fix the validation workflow and make sure that the ingestion-core publish works when we close this

**Comment by pmbrull** [comment]
Also @roeap, I reviewed the `py-tests` execution locally and it is failing for me too, regardless of the `black` issue. I believe it is because we are missing the class renaming in the files under `ingestion/src/airflow_provider_openmetadata`.

@harshach, we could either update it here or give it a go after closing this.

Thanks both!

**Comment by roeap** [comment]
@pmbrull - I went ahead and fixed the remaining imports. When going through the files I noticed one issue I could not resolve. 
In `ingestion/src/metadata/ingestion/source/amundsen.py` we have an import 
```py
from metadata.ingestion.models.user import User
```

However that class does not seem to exists. Looking at the code it also does not seem to refer to the generated user model. I could just create a model according to the usage, but as I am not familiar with the codebase I was unsure what the intent is there. Also the other models in the location seem to be names a bit differently - e.g. `MetadataTeam`.

**Comment by roeap** [comment]
@pmbrull @harshach - I finally set up a local test environment to investigate the test failures a bit more. Turns out the changes ran a lot deeper then I expected. In certain tests we use the python dunder methods quite heavily to generate scenarios. Specifically it seems that certain naming conventions were assumed in the tests, which I changed. The cases I changed were related to `Create***EntityRequest` now being named `Create***Request`.

**Comment by pmbrull** [comment]
> @pmbrull @harshach - I finally set up a local test environment to investigate the test failures a bit more. Turns out the changes ran a lot deeper then I expected. In certain tests we use the python dunder methods quite heavily to generate scenarios. Specifically it seems that certain naming conventions were assumed in the tests, which I changed. The cases I changed were related to `Create***EntityRequest` now being named `Create***Request`.
> 
> In any case we finally have passing tests - yay :).

Nice catch @roeap 🚀 💪  amazing job here!!

**Comment by pmbrull** [comment]
> @pmbrull - I went ahead and fixed the remaining imports. When going through the files I noticed one issue I could not resolve. In `ingestion/src/metadata/ingestion/source/amundsen.py` we have an import
> 
> ```python
> from metadata.ingestion.models.user import User
> ```
> 
> However that class does not seem to exists. Looking at the code it also does not seem to refer to the generated user model. I could just create a model according to the usage, but as I am not familiar with the codebase I was unsure what the intent is there. Also the other models in the location seem to be names a bit differently - e.g. `MetadataTeam`.

Hi @roeap, those classes are deprecated in favour of the ones derived from the JSON schemas. We must have missed them when making the update. @harshach, I'll open an issue to fix that, but otherwise, I think we can merge this in. Thanks!

**Review by harshach — APPROVED** [review]

**Pull request merged**
